video/out/gpu/video.c

```diff
@@ -2305,7 +2303,7 @@ static void pass_read_video(struct gl_video *p)
         struct scaler *scaler = &p->scaler[scaler_id];
 
         // bilinear scaling is a free no-op thanks to GPU sampling
-        if (strcmp(conf->kernel.name, "bilinear") != 0) {
+        if (!conf->kernel.name || strcmp(conf->kernel.name, "bilinear") != 0) {
```

Comment:
if `!name` doesn't that mean whatever is in `conf` is invalid?

Reply:
Yes, but `reinit_scaler` in the `pass_sample` call will reinitialized all the scaler and conf stuff.

Reply:
Seems a bit messy to have different code work in different ways because it either handles the fallback or doesn't, but it looks like haasn wants to rewrite this code anyway.